### PR TITLE
Fix #100047

### PR DESCRIPTION
**Pull request by cgocast**
We remove TaintedSql sink for PDOStatement related methods

### CODE REVIEW

File: stubs/extensions/pdo.phpstub [hunk 1]
```diff
@@ -111,15 +111,11 @@ class PDO
     public function lastInsertId(?string $name = null) {}
 
     /**
-     * @psalm-taint-sink sql $query
```
Comment (danog) [inline stubs/extensions/pdo.phpstub:114]: This is wrong, the query passed to prepare is still a taint sink.

File: stubs/extensions/pdo.phpstub [hunk 2]
```diff
@@ -111,15 +111,11 @@ class PDO
...
-     *
      * @return PDOStatement|false
      */
     public function prepare(string $query, array $options = []) {}
 
     /**
-     * @psalm-taint-sink sql $query
```
Comment (danog) [inline stubs/extensions/pdo.phpstub:121]: Same here

File: stubs/extensions/pdo.phpstub [hunk 3]
```diff
@@ -150,16 +146,6 @@ class PDOStatement implements Traversable
      * @return false|T
      */
     public function fetchObject($class = \stdclass::class, array $ctorArgs = array()) {}
-
-    /**
-     * @psalm-taint-sink sql $value
-     */
-    public function bindValue(string|int $param, mixed $value, int $type = PDO::PARAM_STR): bool {}
```
Comment (danog) [inline stubs/extensions/pdo.phpstub:157]: Don't remove the methods please, just remove the taint sink annotation.

File: stubs/extensions/pdo.phpstub (hunk 1 above)
Comment (cgocast) [reply stubs/extensions/pdo.phpstub]: Done !

File: stubs/extensions/pdo.phpstub (hunk 2 above)
Comment (cgocast) [reply stubs/extensions/pdo.phpstub]: Done !

File: stubs/extensions/pdo.phpstub (hunk 3 above)
Comment (cgocast) [reply stubs/extensions/pdo.phpstub:157]: Done !

**Comment by danog** [comment]
LGTM, @orklah you agree?

**Comment by orklah** [comment]
Thanks!

**Pull request merged**